src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -37,7 +42,26 @@ export const UserAvatar: React.FC<UserAvatarProps> = ({
     "user-avatar--clickable": onClick !== undefined,
   });
 
+  const isOnline = recentWorldUsers.find(
+    (worldUser) => worldUser.id === user?.id
+  );
+
+  const status = user?.status ?? "";
+
   const imageClasses = classNames("user-avatar__image", imageClassName);
+  const statusClasses = classNames("user-avatar__status-dot", {
+    [`user-avatar__status-dot--${status}`]: status,
+    "user-avatar__status-dot--large": large,
+  });
+  const statusOfflineClasses = classNames(
+    "user-avatar__status-dot",
+    "user-avatar__status-dot--offline",
+    { "user-avatar__status-dot--large": large }
+  );
+
+  const statusComponent = isOnline
+    ? showStatus && status && <span className={statusClasses} />
+    : showStatus && <span className={statusOfflineClasses} />;
```

Comment:
This must be rewritten into one classNames object.

smth like:

```
classNames("user-avatar__status-dot", {
"user-avatar__status-dot--offline": !isOnline,
"user-avatar__status-dot--online": isOnline,
[`user-avatar__status-dot--${status}`]: isOnline && status,
"user-avatar__status-dot--large": large,  });
```

And not display dot if it's offline. Using css

Reply:
Thanks!

Reply:
https://github.com/sparkletown/sparkle/pull/1384/commits/119d07836ecd2582c9bd23a81d188b3e93688825